src/stringobj.h

```diff
@@ -296,16 +356,22 @@ Obj ConvImmString(Obj str);
 **
 ** The cstring is assumed to be allocated on the heap, hence its length
 ** is dynamic and must be computed during runtime using strlen.
+**
+** This macro is provided for backwards compatibility of packages.
+** MakeString and MakeImmString are as efficient and should be used instead.
```

Comment:
The only package using `C_NEW_STRING_DYN` is `profiling`. So if you changed that in profiling (which is of course trivial to do, even if you want to retain compatibility with GAP 4.8... :-), you could remove `C_NEW_STRING_DYN` completely.

But of course this can also happen in a later PR.